source/sliding_puzzle.py

```diff
@@ -37,31 +48,35 @@ def __bool__(self) -> bool:
     @property
     def data(self) -> List[str]:
         """List of lines of the image."""
-        return self._data or [' ' * self.width] * self.height
+        return self._data or [" " * self.width] * self.height
 
     @property
-    def empty(self) -> bool:
+    def is_empty(self) -> bool:
```

Comment:
It's just a convention I'm aware of for property that does this kind of computation, it's just to distinguish them from pure class attribute, I can change it back

Reply:
Not really, properties are more attributes than functions (from the user's perspectivez not the source's). If we'd make this a function it would be renamed `is_empty()` yeah, but now it's confusing to have an attribute with a verb in it.

The computation itself is next to nothing, we just check if the list is empty.

Reply:
yeah you are right, actually I thought I removed the property decorator